File: MosPosudit.Model/Requests/Review/ReviewInsertRequest.cs
using System.ComponentModel.DataAnnotations;

namespace MosPosudit.Model.Requests.Review
{
    public class ReviewInsertRequest
    {
        [Required(ErrorMessage = "Tool ID is required")]
        public int ToolId { get; set; }

        [Required(ErrorMessage = "Rental ID is required")]
        public int RentalId { get; set; }

        [Required(ErrorMessage = "Rating is required")]
        [Range(1, 5, ErrorMessage = "Rating must be between 1 and 5")]
        public int Rating { get; set; }

        [StringLength(1000, ErrorMessage = "Comment cannot be longer than 1000 characters")]
        public string? Comment { get; set; }

        // UserId will be set from authenticated user context
        public int UserId { get; set; }
    }
}

File: MosPosudit.UI/shared/lib/dtos/review/review_insert_request.dart
class ReviewInsertRequestDto {
  final int toolId;
  final int rating;
  final String? comment;

  ReviewInsertRequestDto({
    required this.toolId,
    required this.rating,
    this.comment,
  });

  Map<String, dynamic> toJson() => {
        'toolId': toolId,
        'rating': rating,
        'comment': comment,
      };
}

